src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -530,6 +516,15 @@ internal async Task Open(NpgsqlTimeout timeout, bool async, CancellationToken ca
             }
             catch (Exception e)
             {
+                if (canRetry)
+                {
+                    Debug.Assert(SslMode == SslMode.Prefer);
+                    canRetry = false;
+                    SslMode = SslMode.Disable;
+                    Cleanup();
```

Comment:
Note that we haven't actually had a case where we cleanup and then re-open like here - just pointing it out.